src/engine/v3/V3Interpreter.v3

```diff
@@ -1196,6 +1200,32 @@ component V3Interpreter {
 		if (t.reason != TrapReason.NONE) trap(t.reason);
 		else write(t.result, unbox(val));
 	}
+	def doStoreLane<T>(size: byte, log2_size: u3, view: u64 -> T, write: (Range<byte>, T) -> void) {
+		var v = Values.v_v128(pop());
+		// Decode memarg
+		var memarg = codeptr.read_MemArg();
+		var memory = frame.func.instance.memories[memarg.memory_index];
+		var index = popm(memory);
+		if (!u32.?(index)) return trap(TrapReason.MEM_OUT_OF_BOUNDS);
+		if (!u32.?(memarg.offset)) return trap(TrapReason.MEM_OUT_OF_BOUNDS);
+		// Decode immediate
+		var idx = codeptr.read1();
+		// Extract lane
+		var low = v.0, high = v.1;
+		var val: T;
+		def half_lanes = 8 >> log2_size;
+		if (idx < half_lanes) { // Extract a lane from low
+			var shift = u6.view(idx << u3.+(log2_size, 3));
+			val = view(low >> shift);
+		} else { // Extract a lane from high
+			var shift = u6.view((idx - half_lanes) << u3.+(log2_size, 3));
+			val = view(high >> shift);
+		}
+		// Write to memory
+		var t = memory.range_oil_32(u32.!(memarg.offset), u32.!(index), size);
```

Comment:
You should be able to use `memory.range_oil_64` here and avoid the `u32` checks above. Then this routine will be ready for the `memory64` wasm extension.

Reply:
Then I guess these two checks could be also removed?
```
		if (!u32.?(index)) return trap(TrapReason.MEM_OUT_OF_BOUNDS);
		if (!u32.?(memarg.offset)) return trap(TrapReason.MEM_OUT_OF_BOUNDS);
```

Reply:
Yep.